### PR TITLE
Stub v2 sdk functions

### PR DESCRIPTION
**Pull request by sentilesdal**
- stub buyYieldTokens
- stub tradePrincipalTokens
- add more stubbed functions

### CODE REVIEW

File: packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  BigNumber,
+  BigNumberish,
+  ContractReceipt,
+  ContractTransaction,
+  Overrides,
+  Signer,
+} from "ethers";
+
+// this is likely to get a lot more complicated with swap kinds, exact in, exact out etc
+export function calculateTradePrincipalTokens(
```
Comment (DannyDelott) [inline packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.ts:11]: doc block here, probably good to explain that the arrays must be same length

File: packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.test.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { expect } from "chai";
+
+import { calculateTradePrincipalTokens } from "src/calculateTradePrincipalTokens/calculateTradePrincipalTokens";
+
+describe("calculateTradePrincipalTokens", () => {
+  it("Should return a string", async () => {
+    const result = await calculateTradePrincipalTokens(["1", "1"], ["1", "1"]);
```
Comment (DannyDelott) [inline packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.test.ts:7]: this isn't an async function, no need for await

File: packages/core-v2-sdk/src/redeemLiquidity/redeemLiquidity.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  ContractReceipt,
+  ContractTransaction,
+  Overrides,
+  Signer,
+} from "ethers";
+
+// redeem liquidity after the term is mature to the underlying
+export async function redeemLiquidity(
+  amount: BigNumberish,
+  poolAddress: string,
```
Comment (DannyDelott) [inline packages/core-v2-sdk/src/redeemLiquidity/redeemLiquidity.ts:13]: nit: consistent arg order with other SDK methods, should be `(poolAddress, amount, signer, overrides)`

File: packages/core-v2-sdk/src/withdrawLiquidity/withdrawLiquidity.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  ContractReceipt,
+  ContractTransaction,
+  Overrides,
+  Signer,
+} from "ethers";
+
+// withdraw liquidity before the term is expired to underlying and pts
+export async function withdrawLiquidity(
+  amount: BigNumberish,
+  poolAddress: string,
```
Comment (DannyDelott) [inline packages/core-v2-sdk/src/withdrawLiquidity/withdrawLiquidity.ts:13]: same nit re: arg order

File: packages/core-v2-sdk/src/provideLiquidity/provideLiquidity.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import {
+  BigNumber,
+  BigNumberish,
+  ContractReceipt,
+  ContractTransaction,
+  Overrides,
+  Signer,
+} from "ethers";
+
+export async function provideLiquidity(
```
Comment (DannyDelott) [inline packages/core-v2-sdk/src/provideLiquidity/provideLiquidity.ts:10]: doc block

File: packages/core-v2-sdk/src/withdrawLiquidity/withdrawLiquidity.ts (hunk 4 above)
Comment (DannyDelott) [reply packages/core-v2-sdk/src/withdrawLiquidity/withdrawLiquidity.ts:13]: and doc block

**Review by DannyDelott — APPROVED** [review]
handful of small thigns

File: packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.ts (hunk 1 above)
Comment (sentilesdal) [reply packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.ts:11]: I'm not sure what u mean by one-stop shop.  yeah only traded against base asset, specifically in our CC Pools, so maybe something like `calcSwapConvergentCurvePool` or something would make the most sense.

**Pull request merged**